### DISCUSSION TITLE
erro no codigo

### DISCUSSION THREAD

**Discussion by Burudoggu** [dgadelha/Portugol-Webstudio#345] [Dúvidas / Perguntas] [answered]
me ajudem pfv a arrumar esse codigo
![algoritmo2](https://github.com/user-attachments/assets/d94ae28a-1afd-4c35-bd01-4a5fd51b08d4)
![erro1](https://github.com/user-attachments/assets/a5a898c1-d549-4d65-b14e-259556aaf432)
![erro2](https://github.com/user-attachments/assets/c663f897-1f0c-450d-adb8-4579f65556e5)

**Comment by WilCb** [comment, marked as answer]
Encontrei alguns erros no código e na lógica, o `se`, o `senao` e o `enquanto`, estão sem fechamento e abertura de chaves `{ }`, além de que o "enquanto" é um tipo de estrutura de repetição, já o `faca` é usado de outra forma, onde você primeiro coloca o `faca` e no final no bloco coloca o `enquanto(condição de parada)` .

### Ajustei as variáveis para ficar mais legível:

`inteiro base, expoente, potencia = 1, i = 1`

A variável ` i ` não precisa receber valor fora do se apenas durante sua declaração e dentro do `enquanto` para que o loop tenha um fim, você pode colocar `i++`, pois é o mesmo que `i = i + 1`, ou seja, o `i++` incrementa mais 1 a variável.

Além disso o `escreva("O resultado de ", base, " elevado a ", expoente, " é: ", potencia)` precisa ficar dentro da condicional `se` pois ele só será mostrado se a condição for verdadeira caso o contrário vai para o 
`
senao {
      escreva("Expoente deve ser um número inteiro positivo!")
 }
`

### Segue o código completo: 

```portugol
programa {
  funcao inicio() {
    inteiro base, expoente, potencia = 1, i = 1

    // Obtenção de valores das variáveis pelo usuário
    escreva("Digite o valor da base: ")
    leia(base)
    escreva("Digite o valor do expoente: ")
    leia(expoente)

    // O se tem abertura e fechamento com chaves {}
    se(expoente > 0) {
      enquanto(i <= expoente) {
        potencia = potencia * base
        i++
      }
      escreva("O resultado de ", base, " elevado a ", expoente, " é: ", potencia)
    }

    // O senao tbm precisa conter abertura e fechamento com chaves {}
    senao {
      escreva("Expoente deve ser um número inteiro positivo!")
    }
  }
}
```

**Comment by jhoni1517** [comment]
alguém para me ajudar, quero fazer um sistema que peça valor de X, depois leia e paça o valor de Y e depois calcule uma divisão de xpor Y, gostaria de fazer assim para trabalhar com variáveis, mas não sei onde to errando 

```portugol
programa {
  funcao inicio() {
   
   cadeia Resto_da_Divisao
   real Resto_da_Divisao
   real X
   real Y
  
    escreva("Digite o valor de X: ")
    leia(X)
    se (X<>0) entao 
  
   
    escreva("Digite o valor de Y: ")
    leia(Y)

   se (Y > 0) entao

    Resto_da_Divisao = (X/Y) 

        escreva("O resultado da divisão de ", X, " por ", Y, " é: ", Resto_da_Divisao, "\n")

    senao
        escreva("Erro: divisão por zero não é permitida!\n")


    Resto_da_Divisao = (X/Y)
  }
}
```

**Reply by WilCb** [reply]
# Você devia criar um novo tópico para esta pergunta, mas vou te ajudar

1. As variáveis devem começar com letras minusculas, e ou vc usa "_" para separar_cada_palavra chamado de Snake case ou você usa Camel case onde a primeira palavra começa com minuscula e as proximas com maiusculas "exemploDeCamelCase". Segue um link para explicar melhor [Convenções de nomenclatura: Camel, Pascal, Kebab e Snake case](https://www.alura.com.br/artigos/convencoes-nomenclatura-camel-pascal-kebab-snake-case?srsltid=AfmBOoqKiqNcX3bqlLfhFBbSrupcYj_tKAlM15PtFGbut3uhR_Du50Zr)
2. O se não tem a palvra "então" depois
3. Se você quer usar algo para "x diferente de 0", você não usa "<>", tem que usar "x != 0". Segue o código abaixo corrigido

```portugol
programa {
  funcao inicio() {
    real divisao, x, y

    escreva("Digite o valor de X: ")
    leia(x)
    escreva("Digite o valor de Y: ")
    leia(y)

    se (y > 0) {
      divisao = (x / y)
      escreva("O resultado da divisão de ", x, " por ", y, " é: ", divisao, "\n")
    }
    senao {
      escreva("Erro: divisão por zero não é permitida!\n")
    }
    divisao = ( x / y )
  }
}
```

Espero ter ajudado :)
